tests/unit/libexpr/trivial.cc

```diff
@@ -177,6 +177,48 @@ namespace nix {
             )
     );
 
+#define X_EXPAND_IF0(k, v) k "." v
```

Comment:
I have a hard time figuring out what these #defines are expanding to. Maybe you can add a comment explaining what this does?